django/db/models/query.py

```diff
@@ -165,7 +165,7 @@ def __iter__(self):
                 annotation_fields,
             ) = self.queryset.resolve_model_init_order()
             model_cls = self.queryset.model
-            if model_cls._meta.pk.attname not in model_init_names:
+            if {f.attname for f in model_cls._meta.pk_fields} - set(model_init_names):
```

Comment:
```suggestion
            if any(f.attname not in model_init_names for f in model_cls._meta.pk_fields):
```
Maybe? I think I find this easier to read